### PR TITLE
Montana Standard Deduction

### LINKED ISSUE #2327: Montana Standard Deduction
(b)
https://leg.mt.gov/bills/mca/title_0150/chapter_0300/part_0210/section_0310/0150-0300-0210-0310.html

### PR DESCRIPTION
**Pull request by Yinan0409**
### Summary
:sparkles::chart_with_upwards_trend::pencil2:

This pull request adds support for Montana standard deduction in the policy engine, by introducing a new parameter, variable and tests. It also updates the payroll tax revenue data and fixes a typo in a Virginia test file.

> _`Montana` deduction_
> _New parameter and tests_
> _Fall leaves change with code_

### Walkthrough
*  Add a new variable, parameter and tests for Montana standard deduction ([link](https://github.com/PolicyEngine/policyengine-us/pull/2366/files?diff=unified&w=0#diff-791439f2ea4076d18fc83b3764495d4232528361b06768a2488cbde52cc7ccffR1-R21), [link](https://github.com/PolicyEngine/policyengine-us/pull/2366/files?diff=unified&w=0#diff-33ab6e73632e3875edf292f8c668473d3b29c960ba6532211b7bb835aa690a4fR1-R15), [link](https://github.com/PolicyEngine/policyengine-us/pull/2366/files?diff=unified&w=0#diff-f654caf09004c5f0dd723a999e8df395145de12f3004aa5a82c0a25fa3a35039R1-R31))
* Update the historical and projected revenue data for payroll tax ([link](https://github.com/PolicyEngine/policyengine-us/pull/2366/files?diff=unified&w=0#diff-878ad26cac31800ea605ffc8242a768ef868c3ffadb99a214c72a8703ccd0251L13-R13), [link](https://github.com/PolicyEngine/policyengine-us/pull/2366/files?diff=unified&w=0#diff-4c9f860119108430210fbac152dfc14bc12ff0594be9b1578da12fdda4838d39L18-R18))
* Fix a typo in the comment of the test file for Virginia federal and state employees subtraction ([link](https://github.com/PolicyEngine/policyengine-us/pull/2366/files?diff=unified&w=0#diff-f0625e8d5893abd5afe28b8738cdc648508a3676c9a1a563f52626c5ce188833L78-R78))
* Add a new entry to the changelog ([link](https://github.com/PolicyEngine/policyengine-us/pull/2366/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))

fixes #2327

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    fixed:
+    - Parameter, variable and tests for Montana Standard Deduction (b).
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Montana Standard Deduction.
```

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    fixed:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:3]:
```suggestion
    added:
```

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+description: MT provides filers a maximum standard deduction of this amount, depending on filing status.
+metadata:
+  label: MT standard deduction
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml:3]:
```suggestion
  label: Montana standard deduction
```

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+description: MT provides filers a maximum standard deduction of this amount, depending on filing status.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml:1]:
```suggestion
description: Montana provides filers a maximum standard deduction of this amount, depending on filing status.
```

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml:1]: need to add [AGI part](https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/05/Montana-Idividiual-Income-Tax-Return-Form-2-2022v6.2.pdf)

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+from policyengine_us.model_api import *
+
+
+class mt_standard_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "MT standard deduction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:7]:
```suggestion
    label = "Montana standard deduction"
```

**Review by PavelMakarchuk — COMMENTED** [review]
need to add AGI part

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+description: Montana provides filers a maximum standard deduction of this amount, depending on filing status.
+metadata:
+  label: Montana standard deduction max amount
+  period: year
+  unit: currency-USD
+  breakdown:
+    - filing_status
+  reference:
+    - title: MT Code, Income Tax - Deductions, § 42-15-512 (2)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml:9]:
```suggestion
    - title: MT Code, Income Tax - Deductions, § 42-15-523 (2)
```

Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml:9]: The values here are not in this reference. Please add tax form references to ensure all values have a reference. Same for min.

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/rate.yaml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+description: Montana multiplies the agi by this rate under the standard deduction.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/rate.yaml:1]:
```suggestion
description: Montana provides a standard deduction equal to this percentage of Montana adjusted gross income, subject to minimum and maximum values.
```

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    entity = TaxUnit
+    label = "Montana standard deduction"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(period).gov.states.mt.tax.income.deductions.standard
+        agi = tax_unit("adjusted_gross_income", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:15]: needs to be Montana AGI `mt_agi`

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+        # standard deduction is a percentage of AGI that
+        # is bounded by a min/max by filing status.
+        min_amount = max_((p.rate * agi), p.min[filing_status])
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:18]:
```suggestion
        min_amount = p.min[filing_status]
        max_amount = p.max[filing_status]
        unbounded = p.rate * agi
        return max_(min_(unbounded, max_amount), min_amount)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    entity = TaxUnit
+    label = "Montana standard deduction"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(period).gov.states.mt.tax.income.deductions.standard
+        agi = tax_unit("mt_agi", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:15]: need to create mt_agi.py

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+  label: Montana standard deduction max amount
+  period: year
+  unit: currency-USD
+  breakdown:
+    - filing_status
+  reference:
+    - title: MT Code, Income Tax - Deductions, § 42-15-523 (2)
+      href: https://rules.mt.gov/gateway/RuleNo.asp?RN=42%2E15%2E523
+    - title: Montana Tax Form 2022 (Page 7)
+      href: https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/05/Montana-Idividiual-Income-Tax-Return-Form-2-2022v6.2.pdf
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml:12]:
```suggestion
      href: https://mtrevenue.gov/wp-content/uploads/dlm_uploads/2023/05/Montana-Idividiual-Income-Tax-Return-Form-2-2022v6.2.pdf#page=7
```

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+metadata:
+  label: Montana standard deduction max amount
+  period: year
+  unit: currency-USD
+  breakdown:
+    - filing_status
+  reference:
+    - title: MT Code, Income Tax - Deductions, § 42-15-523 (2)
+      href: https://rules.mt.gov/gateway/RuleNo.asp?RN=42%2E15%2E523
+    - title: Montana Tax Form 2022 (Page 7)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml:11]: Specify page numbers for pdf files in the URL, and sections / line numbers in the titles. Please apply to others as well.
```suggestion
    - title: Montana Tax Form 2022 | Standard Deduction | Line 3
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/rate.yaml [hunk 13]
```diff
@@ -0,0 +1,10 @@
+description: Montana provides a standard deduction equal to this percentage of Montana adjusted gross income, subject to minimum and maximum values.
+values:
+  2021-01-01: 0.2
+metadata:
+  period: year
+  unit: /1
+  label: Montana standard deduction rate
+  reference:
+  - title: MT Code, Income Tax - Deductions, § 42-15-512 (2)
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/rate.yaml:9]: need to add 2021 and 2022 tax form references

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+        # standard deduction is a percentage of AGI that
+        # is bounded by a min/max by filing status.
+        min_amount = p.min[filing_status]
+        max_amount = p.max[filing_status]
+        unbounded = p.rate * agi
+        return max_(min_(unbounded, max_amount), min_amount)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:21]:
```suggestion
        max_amount = min_(unbounded, max_amount)
        return max_(max_amount, min_amount)
```

File: policyengine_us/variables/gov/states/mt/tax/income/agi/mt_agi.py [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+class mt_agi(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Montana Adjusted Gross Income"
+    unit = USD
+    definition_period = YEAR
+    reference = ""
+    defined_for = StateCode.MT
+
+    # def formula(tax_unit, period, parameters):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/agi/mt_agi.py:13]: Revert to draft if it's not ready to merge, as this commented code indicates

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/min.yaml [hunk 16]
```diff
@@ -0,0 +1,30 @@
+description: Montana provides a standard deduction of up to this amount.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/min.yaml:1]:
```suggestion
description: Montana provides a minimum standard deduction of this amount, based on filing status.
```

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+
+class mt_standard_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Montana standard deduction"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:12]: need to add condition that the filer does not itemizes, use the variable: tax_unit_itemizes

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+        filing_status = tax_unit("filing_status", period)
+        p = parameters(period).gov.states.mt.tax.income.deductions.standard
+        agi = tax_unit("mt_agi", period)
+        # standard deduction is a percentage of AGI that
+        # is bounded by a min/max by filing status.
+        min_amount = p.min[filing_status]
+        max_amount = p.max[filing_status]
+        unbounded = p.rate * agi
+        max_amount = min_(unbounded, max_amount)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:21]:
```suggestion
        deduction_amount = min_(unbounded, max_amount)
```

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+        return max_(max_amount, min_amount)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:22]:
```suggestion
        return max_(deduction_amount, min_amount)
```

File: policyengine_us/variables/gov/states/mt/tax/income/agi/mt_agi.py [hunk 20]
```diff
@@ -0,0 +1,17 @@
+from policyengine_us.model_api import *
+
+
+class mt_agi(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Montana Adjusted Gross Income"
+    unit = USD
+    definition_period = YEAR
+    reference = ""
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/mt/tax/income/agi/mt_agi.py:10]:
```suggestion
```

File: policyengine_us/tests/policy/baseline/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.yaml [hunk 21]
```diff
@@ -0,0 +1,44 @@
...
+- name: $8,000 for widow. #0.2*8000=1600 (lower than min)
+  period: 2021
+  input:
+    state_code: MT
+    mt_agi: 8_000
+    filing_status: WIDOW
+  output:
+    mt_standard_deduction: 2_140
+
+- name: $30,000 for seperate. #0.2*30000=6000 (higher than max)
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.yaml:19]:
```suggestion
- name: $30,000 for separate. #0.2*30000=6000 (higher than max)
```

File: policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml [hunk 22]
```diff
@@ -0,0 +1,31 @@
+description: Montana provides filers a maximum standard deduction of this amount, depending on filing status.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/mt/tax/income/deductions/standard/max.yaml:1]:
```suggestion
description: Montana provides this maximum standard deduction amount, based on filing status.
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py [hunk 23]
```diff
@@ -0,0 +1,24 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Montana standard deduction"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.MT
+
+    def formula(tax_unit, period, parameters):
+        filing_status = tax_unit("filing_status", period)
+        itemizing = tax_unit("tax_unit_itemizes", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/mt/tax/income/deductions/standard/mt_standard_deduction.py:14]: Remove this condition - we check the itemization choice in a later step in the computation tree so we can optimize the filer's choice.

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**